### PR TITLE
Links between DOWNLOADS.md and TESTING.md

### PR DESCRIPTION
**Pull request by slook**
+ Added: Link to TESTING.md, because this makes it easier to navigate around the website.
+ Added: Link to DOWNLOADS.md, incase website visitors get stuck on the wrong page.
+ Added: Link to DEPENDENCIES.md, this is useful and relevant in TESTING.md.

- Changed: Clarify "current stable version" and "latest unstable build" terminology to be more user-friendly.
- Changed: "Download Nicotine+" links to include SEO and user-friendly keywords.
- Changed: Plain text to hyperlinks for Flatpak, pip and Git, incase these tools need setup or config on the platform.
- Removed: Paragraph about source code from DOWNLOADS, it was an irrelevant statement of the obvious.

### CODE REVIEW

File: doc/DOWNLOADS.md [hunk 1]
```diff
@@ -130,3 +130,7 @@ Once Homebrew is set up, run the following:
 ```sh
 brew install nicotine-plus
 ```
+
+## Testing Nicotine+
+
+If you want to download unstable builds and help test Nicotine+, see [TESTING.md](TESTING.md).
```
Comment (mathiascode) [inline doc/DOWNLOADS.md:136]: I'd prefer if this was at the top, under the "Download Nicotine+" heading.

File: doc/TESTING.md [hunk 2]
```diff
@@ -80,3 +82,7 @@ Portable packages are also available. They can be run from your home directory,
 Unstable macOS installers are generated after every commit to the master branch, and should only be used for testing.
 
 - [Download Installer](https://nightly.link/nicotine-plus/nicotine-plus/workflows/packaging/master/macos-installer.zip)
+
+## Stable builds
+
+If you want to download the current stable version of Nicotine+, see [DOWNLOADS.md](DOWNLOADS.md).
```
Comment (mathiascode) [inline doc/TESTING.md:88]: Same here.

File: doc/TESTING.md [hunk 3]
```diff
@@ -80,3 +66,22 @@ Portable packages are also available. They can be run from your home directory,
 Unstable macOS installers are generated after every commit to the master branch, and should only be used for testing.
 
 - [Download Installer](https://nightly.link/nicotine-plus/nicotine-plus/workflows/packaging/master/macos-installer.zip)
+
+## Git
```
Comment (mathiascode) [inline doc/TESTING.md:70]: If you move this section out, pip should also be moved.

I would revert this change, since setting up an environment on Windows/macOS is rather annoying/time-consuming. The Windows/macOS-specific builds are better options.

File: doc/DOWNLOADS.md [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 
-Download stable builds of Nicotine+ for your operating system.
+Download stable builds of Nicotine+ for your operating system. For the release notes, see [NEWS.md](/NEWS.md).
+
+If you want to download unstable builds and help test Nicotine+, see [TESTING.md](TESTING.md).
 
 ## GNU/Linux, *BSD, Solaris
 
-If you have no need to modify the Nicotine+ source, you are strongly recommended to use packages for your distribution/operating system. This will save you time.
+If packages are available for your distribution/operating system, this will save you time.
```
Comment (mathiascode) [inline doc/DOWNLOADS.md:9]: I don't think we need this line.

File: doc/DOWNLOADS.md [hunk 5]
```diff
@@ -60,33 +62,33 @@ pkg install py-nicotine-plus
...
 
-NOTE: The installer format has changed in Nicotine+ 3.2.0. If you are upgrading from Nicotine+ 3.1.1 or earlier, please uninstall Nicotine+ first (this will not remove your existing settings).
+*NOTE: The installer format has changed since Nicotine+ 3.2.0. If you are upgrading from Nicotine+ 3.1.1 or earlier, please uninstall Nicotine+ first (this will not remove your existing settings).*
 
-- [64-bit Installer](https://github.com/nicotine-plus/nicotine-plus/releases/latest/download/windows-x86_64-installer.zip)  [[SHA256](https://github.com/nicotine-plus/nicotine-plus/releases/latest/download/windows-x86_64-installer.zip.sha256)]
-- [32-bit Installer](https://github.com/nicotine-plus/nicotine-plus/releases/latest/download/windows-i686-installer.zip)  [[SHA256](https://github.com/nicotine-plus/nicotine-plus/releases/latest/download/windows-i686-installer.zip.sha256)]
+- [Download Nicotine+ 64-bit Windows Installer *(current stable version)*](https://github.com/nicotine-plus/nicotine-plus/releases/latest/download/windows-x86_64-installer.zip)  [[SHA256](https://github.com/nicotine-plus/nicotine-plus/releases/latest/download/windows-x86_64-installer.zip.sha256)]
```
Comment (mathiascode) [inline doc/DOWNLOADS.md:85]: Do we need this verbosity? These links are rather awkward to look at and read at this length:
Screenshot from 2022-02-02 21-23-44

I think including "Download" in the original text would be fine, e.g. "Download 64-bit Installer".

Comment (slook) [reply doc/DOWNLOADS.md:85]: But then there would nothing to index with the Windows keyword on the SERP's.

I suggest we don't need _(current stable version)_ that is what causes the awkward appearance.

File: doc/DOWNLOADS.md [hunk 6]
```diff
@@ -30,9 +30,11 @@ sudo dnf install nicotine+
 
 ### Flatpak
 
-If your distribution supports Flatpak, you can install Nicotine+ from Flathub.
+If your distribution supports [Flatpak](https://www.flatpak.org/setup/), you can install the current stable version of [Nicotine+ from Flathub](https://flathub.org/apps/details/org.nicotine_plus.Nicotine), run the following:
 
-- [Download Nicotine+ on Flathub](https://flathub.org/apps/details/org.nicotine_plus.Nicotine)
+```sh
+flatpak install flathub org.nicotine_plus.Nicotine
```
Comment (mathiascode) [inline doc/DOWNLOADS.md:36]: This won't work if the user hasn't added the Flathub repository before.

Comment (slook) [reply doc/DOWNLOADS.md:36]: Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/f02090e6edf9eeff0fab2a5f647c759caf37c558

File: doc/TESTING.md (hunk 3 above)
Comment (slook) [reply doc/TESTING.md]: > If you move this section out, pip should also be moved.

@redactedscribe seems to have a better approach in #2037
